addons/pkg-npm/model-java/src/main/java/org/commonjava/indy/pkg/npm/model/io/VersionMetadataDeserializer.java

```diff
@@ -56,35 +59,52 @@ public VersionMetadata deserialize( JsonParser jsonParser, DeserializationContex
         {
             repoNode = repoNode.get( 0 );
         }
-        vm.setRepository( new Repository( parseValue( repoNode, "type"), parseValue( repoNode, "url") ) );
-
-        vm.setAuthor( parseObject( mapper, vNode.get( "author" ), UserInfo.class ));
+        vm.setRepository( new Repository( parseValue( repoNode, "type" ), parseValue( repoNode, "url" ) ) );
+        vm.setDescription( parseValue( vNode, "description" ) );
+        vm.setMain( parseValue( vNode, "main" ) );
+        vm.setUrl( parseValue( vNode, "url" ) );
+        vm.setHomepage( parseValue( vNode, "homepage" ) );
+        vm.setGitHead( parseValue( vNode, "gitHead" ) );
+        vm.setTitle( parseValue( vNode, "title" ) );
+        vm.setDeprecated( parseValue( vNode, "deprecated" ) );
+        vm.setLib( parseValue( vNode, "lib" ) );
+        vm.setShasum( parseValue( vNode, "_shasum" ) );
+        vm.setFrom( parseValue( vNode, "_from" ) );
+        vm.setNpmVersion( parseValue( vNode, "_npmVersion" ) );
+        vm.setNodeVersion( parseValue( vNode, "_nodeVersion" ) );
+
+        vm.setAuthor( parseObject( mapper, vNode.get( "author" ), UserInfo.class ) );
         vm.setBugs( parseObject( mapper, vNode.get( "bugs" ), Bugs.class ) );
         vm.setDist( parseObject( mapper, vNode.get( "dist" ), Dist.class ) );
         vm.setDirectories( parseObject( mapper, vNode.get( "directories" ), Directories.class ) );
-        vm.setKeywords( parseList( mapper, vNode.get( "keywords" ), String.class ) );
+        vm.setCommitplease( parseObject( mapper, vNode.get( "commitplease" ), Commitplease.class ) );
         vm.setLicense( parseObject( mapper, vNode.get( "license" ), License.class ) );
-        vm.setMain( parseValue( vNode, "main" ) );
-        vm.setUrl( parseValue( vNode, "url" ) );
-
-        vm.setContributors( parseList( mapper, vNode.get( "contributors" ), UserInfo.class ));
-        vm.setEngines( parseList( mapper, vNode.get( "engines" ), Engines.class ));
-
         vm.setDependencies( parseObject( mapper, vNode.get( "dependencies" ), Map.class ) );
         vm.setDevDependencies( parseObject( mapper, vNode.get( "devDependencies" ), Map.class ) );
-        vm.setDependencies( parseObject( mapper, vNode.get( "bin" ), Map.class ) );
+        vm.setBin( parseObject( mapper, vNode.get( "bin" ), Map.class ) );
+        vm.setJsdomVersions( parseObject( mapper, vNode.get( "jsdomVersions" ), Map.class ) );
+        vm.setScripts( parseObject( mapper, vNode.get( "scripts" ), Map.class ) );
+        vm.setEngineSupported( parseObject( mapper, vNode.get( "_engineSupported" ), Boolean.class ) );
+        vm.setNpmUser( parseObject( mapper, vNode.get( "_npmUser" ), UserInfo.class ) );
+        vm.setNpmJsonOpts( parseObject( mapper, vNode.get( "_npmJsonOpts" ), NpmJsonOpts.class ) );
+        vm.setNpmOperationalInternal(
+                parseObject( mapper, vNode.get( "_npmOperationalInternal" ), NpmOperationalInternal.class ) );
```

Comment:
Do we need to include this ? I saw this PR added more fields which starts with "_", I learned that `All properties beginning with _ or $ are also reserved for package registries to use that their discretion. `,  I think it's enough to just include the fields mentioned here: https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md

Reply:
@sswguo, I just see Jan's comment mention the _* fileds, so add them based on the case, you mean don't need to keep all these reserved properties in package.json?

Reply:
@yma96 There are some in above link, I think it's fine to just include those, like `_id` etc.  we can discuss this in the meeting.